examples/cpp-simple/server.cc

```diff
@@ -16,101 +16,115 @@
 // the C++ SDK.
 
-#include <iostream>
-#include <thread>
 #include <agones/sdk.h>
 #include <grpc++/grpc++.h>
+#include <iostream>
+#include <thread>
 
 // send health check pings
 void doHealth(agones::SDK *sdk) {
-    while (true) {
-        if (!sdk->Health()) {
-            std::cout << "Health ping failed" << std::endl;
-        } else {
-            std::cout << "Health ping sent" << std::endl;
-        }
-        std::this_thread::sleep_for(std::chrono::seconds(2));
+  while (true) {
+    if (!sdk->Health()) {
+      std::cout << "Health ping failed" << std::endl;
+    } else {
+      std::cout << "Health ping sent" << std::endl;
     }
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+  }
 }
 
 // watch GameServer Updates
 void watchUpdates(agones::SDK *sdk) {
-    std::cout << "Starting to watch GameServer updates..." << std::endl;
-    sdk->WatchGameServer([](stable::agones::dev::sdk::GameServer gameserver){
-        std::cout << "GameServer Update, name: " << gameserver.object_meta().name() << std::endl;
-        std::cout << "GameServer Update, state: " << gameserver.status().state() << std::endl;
-    });
+  std::cout << "Starting to watch GameServer updates..." << std::endl;
+  sdk->WatchGameServer([](agones::dev::sdk::GameServer gameserver) {
+    std::cout << "GameServer Update, name: " << gameserver.object_meta().name()
+              << std::endl;
+    std::cout << "GameServer Update, state: " << gameserver.status().state()
+              << std::endl;
+  });
 }
 
 int main() {
-    std::cout << "C++ Game Server has started!" << std::endl;
-
-    std::cout << "Getting the instance of the SDK!" << std::endl;
-    agones::SDK *sdk = new agones::SDK();
-
-    std::cout << "Attempting to connect..." << std::endl;
-    bool connected = sdk->Connect();
-    if (!connected) {
-        std::cout << "Could not connect to the sidecar. Exiting!" << std::endl;
-        return -1;
-    }
-    std::cout << "...handshake complete." << std::endl;
-
-    std::thread health (doHealth, sdk);
-    std::thread watch (watchUpdates, sdk);
-
-    std::cout << "Setting a label" << std::endl;
-    grpc::Status status = sdk->SetLabel("test-label", "test-value");
-    if (!status.ok()) {
-        std::cout << "Could not run SetLabel(): "+ status.error_message() + ". Exiting!" << std::endl;
-        return -1;
-    }
-
-    std::cout << "Setting an annotation" << std::endl;
-    status = sdk->SetAnnotation("test-annotation", "test value");
-    if (!status.ok()) {
-        std::cout << "Could not run SetAnnotation(): "+ status.error_message() + ". Exiting!" << std::endl;
+  std::cout << "C++ Game Server has started!" << std::endl;
+
+  std::cout << "Getting the instance of the SDK!" << std::endl;
+  agones::SDK *sdk = new agones::SDK();
+
+  std::cout << "Attempting to connect..." << std::endl;
+  bool connected = sdk->Connect();
+  if (!connected) {
+    std::cout << "Could not connect to the sidecar. Exiting!" << std::endl;
+    return -1;
```

Comment:
Note: With a heap-allocated SDK, this return would leak the SDK. Even though this is main() and we may not care about this leak, leak checkers will care and would report it. So it's best to avoid if possible.

Reply:
ack